Lib/test/test_exceptions.py

```diff
@@ -1695,6 +1695,19 @@ def __getattr__(self, attr):
 
             self.assertIn("blech", err.getvalue())
 
+    def test_attribute_error_with_failing_dict(self):
+        class T:
+            bluch = 1
+            def __dir__(self):
+                raise AttributeError("oh no!")
+
+        try:
+            T().blich
+        except AttributeError as exc:
+            with support.captured_stderr() as err:
+                sys.__excepthook__(*sys.exc_info())
+
+        self.assertNotIn("blech", err.getvalue())
```

Comment:
This should be `"oh no!"` now, right?

Reply:
The "blech" is the name of the attribute for the suggestion, but we can test the exception as well.

Reply:
Done in 34fbac